packages/health/lib/src/health_factory.dart

```diff
@@ -230,6 +263,28 @@ class HealthFactory {
     return success ?? false;
   }
 
+  /// Deletes all records of the given type for a given period of time
+  ///
+  /// Returns true if successful, false otherwise.
+  ///
+  /// Parameters:
+  /// * [type] - the value's HealthDataType
+  /// * [startTime] - the start time when this [value] is measured.
+  ///   + It must be equal to or earlier than [endTime].
+  /// * [endTime] - the end time when this [value] is measured.
+  ///   + It must be equal to or later than [startTime].
+  Future<bool> delete(HealthDataType type, DateTime startTime, DateTime endTime) async {
```

Comment:
in requestAuthorization we verify that we do not have write permission.
How can we delete if we do not have these permissions?

Reply:
Where did you see that we verify that we don't have a write permission?